lib/swiftype-app-search/result_response.rb

```diff
@@ -0,0 +1,35 @@
+# Forwards Ruby Enumerable methods to the :results key of the response for
+# endpoints that contain pagination while still maintaining Hash functionality
+# Methods like #[] and #[]= will still fall back to Hash-like behavior
+
+class SwiftypeAppSearch::ResultResponse
+  include Enumerable
+
+  def initialize(response_json)
+    @response_json = response_json
+
+    if @response_json.is_a?(Hash) && @response_json.key?('errors')
+      raise SwiftypeAppSearch::ClientException, @response_json
+    end
+  end
+
+  attr_reader :response_json
+  alias_method :json, :response_json
+
+  def results
+    return @results if defined?(@results)
+    @results = @response_json['results'] if @response_json.is_a?(Hash)
+  end
+
+  def meta
+    @meta ||= @response_json['meta'].to_h
+  end
+
+  def each
```

Comment:
Super edge case, but still may want to handle it.

Calling `.each` on a plain old array with no arguments looks like it just returns itself.

Calling `.each` on this with no arguments results in 

```
ArgumentError (tried to create Proc object without a block)
```